File: api/routes/avatar.py
"""
头像管理 API 接口
"""

from fastapi import APIRouter, HTTPException, UploadFile, File

from lumen.services import avatar as avatar_service

router = APIRouter()


@router.get("/list")
async def list_avatars():
    return avatar_service.list_avatars()


@router.post("/upload")
async def upload_avatar(file: UploadFile = File(..., description="头像文件")):
    try:
        file_content = await file.read()
        return avatar_service.upload_avatar(file.filename or "", file_content)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"头像上传失败: {str(e)}")


@router.delete("/{avatar_id}")
async def delete_avatar(avatar_id: str):
    try:
        avatar_service.delete_avatar(avatar_id)
        return {"message": f"头像 {avatar_id} 删除成功"}
    except FileNotFoundError:
        raise HTTPException(status_code=404, detail=f"头像不存在: {avatar_id}")
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"头像删除失败: {str(e)}")

File: lumen-Front/src/api/avatar.ts
/**
 * 头像管理 API 客户端
 */
import type { AvatarItem, AvatarUploadResponse } from '../types/avatar';

const API_BASE_URL = 'http://127.0.0.1:8888';

/** GET /avatars/list - 获取所有头像列表 */
export async function listAvatars(): Promise<AvatarItem[]> {
  const res = await fetch(`${API_BASE_URL}/avatars/list`);
  if (!res.ok) {
    const err = await res.json().catch(() => ({ detail: res.statusText }));
    throw new Error(err.detail || `获取头像列表失败: ${res.status}`);
  }
  return res.json();
}

/** POST /avatars/upload - 上传头像 */
export async function uploadAvatar(file: File): Promise<AvatarUploadResponse> {
  const formData = new FormData();
  formData.append('file', file);

  const res = await fetch(`${API_BASE_URL}/avatars/upload`, {
    method: 'POST',
    body: formData,
  });

  if (!res.ok) {
    const err = await res.json().catch(() => ({ detail: res.statusText }));
    throw new Error(err.detail || `上传头像失败: ${res.status}`);
  }
  return res.json();
}

/** DELETE /avatars/:id - 删除头像 */
export async function deleteAvatar(avatarId: string): Promise<{ message: string }> {
  const res = await fetch(`${API_BASE_URL}/avatars/${encodeURIComponent(avatarId)}`, {
    method: 'DELETE',
  });

  if (!res.ok) {
    const err = await res.json().catch(() => ({ detail: res.statusText }));
    throw new Error(err.detail || `删除头像失败: ${res.status}`);
  }
  return res.json();
}
